### PR TITLE
Fix db inconsistencies

### PR DESCRIPTION
**Pull request by Pralish**

### CODE REVIEW

File: app/models/api_resource.rb [hunk 1]
```diff
@@ -37,15 +39,9 @@ def properties_object
   end
 
   def presence_of_required_properties
```
Comment (donrestarone) [inline app/models/api_resource.rb:41]: @Pralish are these validations being tested? I have noticed some validation bugs (eg: when using rich text presence validation doesnt work)

File: app/models/concerns/jsonb_fields_parsable.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Ensure jsonb fields always store hash 
+# Parse JSON string to hash before saving to db if provided value is string
+# Raise validation error if JSON string is not parsable
+
```
Comment (donrestarone) [inline app/models/concerns/jsonb_fields_parsable.rb:4]: @Pralish Nice!

File: app/views/comfy/admin/external_api_clients/_form.html.haml [hunk 3]
```diff
@@ -37,7 +37,7 @@
     = f.text_area :model_definition
 
   #jsoneditor
-  = f.hidden_field :metadata
+  = hidden_field_tag "external_api_client[metadata]", @external_api_client.metadata&.to_json
```
Comment (donrestarone) [inline app/views/comfy/admin/external_api_clients/_form.html.haml:40]: @Pralish no more rails magic??

File: test/fixtures/api_actions.yml [hunk 4]
```diff
@@ -93,10 +93,9 @@ create_api_action_plugin_subdomain_events:
   type: CreateApiAction
   action_type: send_web_request
   include_api_resource_data: true
-  payload_mapping: >
-    {
-      "content": "#{self.representation.body}"
-    }
+  payload_mapping: {
```
Comment (donrestarone) [inline test/fixtures/api_actions.yml:96]: @Pralish what's the change here? We are turning JSON strings into ruby hashes? or is it something else

File: test/models/concerns/json_fields_parsable_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
+require "test_helper"
```
Comment (donrestarone) [inline test/models/concerns/json_fields_parsable_spec.rb:1]: @Pralish can we rename this file to _test?

**Review by donrestarone — CHANGES_REQUESTED** [review]

File: test/fixtures/api_actions.yml (hunk 4 above)
Comment (Pralish) [reply test/fixtures/api_actions.yml:96]: @donrestarone , Yes, we are turning JSON strings into ruby hashes. The '>' was creating json string.

File: test/models/concerns/json_fields_parsable_spec.rb (hunk 5 above)
Comment (Pralish) [reply test/models/concerns/json_fields_parsable_spec.rb]: Sure @donrestarone . Habits😅

**Comment by donrestarone** [comment]
@Pralish @alis-khadka please take a look at [how Im orchastrating my release here](https://github.com/restarone/violet_rails/pull/503)

moving forward, to get our changes on master we will need to: 

1. check if a deployment is in progress
2. if not, create a fresh rc branch from master
3. merge your topic/feature branch to it 
4. wait for deployment
5. test
6. open a PR with RC against master

Let me know if the workflow makes sense!

**Pull request merged**